cloudinit/sources/DataSourceScaleway.py

```diff
@@ -177,51 +186,152 @@ def __init__(self, sys_cfg, distro, paths):
         self.ds_cfg = util.mergemanydict(
             [
                 util.get_cfg_by_path(sys_cfg, ["datasource", "Scaleway"], {}),
-                BUILTIN_DS_CONFIG,
             ]
         )
 
-        self.metadata_address = self.ds_cfg["metadata_url"]
-        self.userdata_address = self.ds_cfg["userdata_url"]
-        self.vendordata_address = self.ds_cfg["vendordata_url"]
-
         self.retries = int(self.ds_cfg.get("retries", DEF_MD_RETRIES))
         self.timeout = int(self.ds_cfg.get("timeout", DEF_MD_TIMEOUT))
+        self.alive = int(self.ds_cfg.get("alive", DEF_MD_ALIVE))
         self._fallback_interface = None
         self._network_config = sources.UNSET
+        self.metadata_urls = DS_BASE_URLS
+        self.userdata_url = None
+        self.vendordata_url = None
+        self.ephemeral_fixed_address = None
+        self.has_ipv4 = True
+        if "metadata_urls" in self.ds_cfg.keys():
+            # Merge and deduplicate lists
+            self.metadata_urls = list(
+                set(self.metadata_urls) | set(self.ds_cfg["metadata_urls"])
+            )
+
+    def _set_metadata_url(self, urls):
+        """
+        Define metadata_url based upon api-metadata URL availability.
+        """
+
+        for url in urls:
```

Comment:
I did check `url_helper.wait_for_url()` late in the work, after creating `_set_urls_on_ip_version()` so it did not seem so useful since we already knew that was the proper IP version.
But after you mentioning it, I went back and its usage makes the code less complex by removing the need for  `_set_urls_on_ip_version()` so I went ahead and used it. 
Thanks for the tip

Reply:
> Thanks for the tip

No problem, and I'm glad this works for you. Delegating dualstack logic in shared code is a win.